### ISSUE TITLE
Apex Legends

### ISSUE THREAD

**Issue by massatt212** [Plagman/gamescope#434]
Gamescope Crashing when trying to load Apex Legends, but works on X11 and Wayland

**Comment by massatt212** [comment]
https://youtu.be/yqnbZ3vVzu4

Idk if its my pc, but this is a small video of whats happening, every other game works except apex

**Comment by rurigk** [comment]
This is happening to me as well

**Comment by pmartiner** [comment]
same over here. here are my logs, in case they're useful https://pastebin.com/raw/r6QKEF8Z

**Comment by madmonki** [comment]
Same here. I tried every command and I got in the game but my inputs were not working.

**Comment by pmartiner** [comment]
> Edit: i don't know why but "gamemoderun" made it to run and without it, game won't launch.

that worked! weird that the game only launches with game mode activated

**Comment by amuprocode645** [comment]
gamemoderun worked for me as well!

**Comment by pmartiner** [comment]
the same happens with fall guys without `gamemoderun`. maybe the issue has to do with the Proton EAC Runtime?

**Comment by madmonki** [comment]
It could be but the annoying thing is after 24 mins a bug kicks in and games start to stutter heavly.

**Comment by pmartiner** [comment]
I've noticed Apex Legends also crashes whenever I add a command prefix (mangohud, as an example), even it's not gamescope. So I think this issue has to do with the EAC Proton Runtime rather than with gamescope

**Comment by pmartiner** [comment]
I've posted more details on [Apex Legend's Proton repo issue](https://github.com/ValveSoftware/Proton/issues/4350)

**Comment by maxxnino** [comment]
> Same here. I tried every command and I got in the game but my inputs were not working. Edit: i don't know why but "gamemoderun" made it to run and without it, game won't launch. Edit2: I didn't know I have to put -e option for steam integration. Input works now.



> > Edit: i don't know why but "gamemoderun" made it to run and without it, game won't launch.
> 
> that worked! weird that the game only launches with `gamemode` activated

I don't think it run `gamemoderun gamescope -w 1920 -h 1080 -r 90 -f -- mangohud %command%`. I still get 144fps and mango hub show dxvk instead of gamescope

**Comment by rezzafr33** [comment]
Today I noticed that the `gamemoderun` trick is no longer working, last night it worked, I've tried in Apex and New World. At least for mangohud removed all `exec` from config, got it working.

**Comment by cyberphantom52** [comment]
Perhaps this will help people getting the game to run or help the devs fix the issue with apex.
The issue is related to LD_PRELOAD, i can successfully run the game with 
`gamescope -- env LD_PRELOAD="$LD_PRELOAD" %command%`

LD_PRELOAD="" doesn't work.

**Comment by cyberphantom52** [comment]
> Perhaps this will help people getting the game to run or help the devs fix the issue with apex. The issue is related to LD_PRELOAD, i can successfully run the game with `gamescope -- env LD_PRELOAD="$LD_PRELOAD" %command%`
> 
> LD_PRELOAD="" doesn't work.

Perhaps @emersion @Joshua-Ashton can shed some light on what may be wrong.

**Comment by pollux56** [comment]
I'm currently getting a similar issue where eac launches, then I hear the game for a couple of seconds then it instantly closes.

Im using

Gamescope -H 1080 -e -f %command% 

Anyone else experienced this issue? 

I have a rx 6700, using KDE plasma 5.27.7

**Comment by sclayton33** [comment]
KDE Plasma 5.27.9
KDE Frameworks 5.111.0
Qt 5.15.11
Kernel 6.6.0-1-MANJARO
Wayland
AMD 6800XT GPU

I'm having this issue as well. It only happens on Wayland, not on X11 or with any other game that I've played on Wayland.

I am unable to get gamescope running with Apex at all in steam. The EAC and game window pops up the first  run then nothing happens after that, just says it's running in steam.

I've tried GE versions of proton but no change.

**Comment by madmonki** [comment]
@sclayton33 I was able to launch the game with gamescope but i got the same behavior same as you on first try. On my first attempt i used `gamescope -H 1080 -e -f -- %command%` . 

Then I launched the game without gamescope to see if my system got another issue and it was working as expected so no issues so far. 

Lastly I tried gamescope again with `gamescope -H 1080 -W 1920 -h 1080 -w 1920 -e -f -- %command%` and it worked. 

I don't know if this will work for you and my blind guess is this is related to game's resolution not being set correctly. That's because I tried some weird resolutions in the past with gamescope for this game and it didn't work at all.
